core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreUtil.java

```diff
@@ -111,4 +117,77 @@ public static Certificate getCertificate(String alias, KeyStore store) throws Ax
         }
     }
 
+    /**
+     * This method checks if the given key store name is for a customkey store by checking the prefix.
+     *
+     * @param keyStoreName      Custom key store name.
+     * @return Boolean value indicating if the given name is for a custom key store.
+     */
+    public static boolean isCustomKeyStore(String keyStoreName) {
+
+        if (keyStoreName.startsWith(RegistryResources.SecurityManagement.CustomKeyStore.KEYSTORE_PREFIX)) {
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * This method builds the QName object for a configuration within the carbon.xml namespace.
+     *
+     * @param localPart     Local part of the configuration.
+     * @return QName object.
+     */
+    public static QName getQNameWithCarbonNS(String localPart) {
+
+        return new QName(ServerConstants.CARBON_SERVER_XML_NAMESPACE, localPart);
+    }
+
+    /**
+     * This method returns the OMElement object corrosponding to the custom key store configuration in Carbon.xml file.
+     *
+     * @param keyStoreName          Name of the custom key store.
+     * @param serverConfiguration   ServerConfigurationService instance.
+     * @return OMElement jobject for the requested configuration.
+     * @throws CarbonException      If failed to retrive the requested configuration.
+     */
+    public static OMElement getCustomKeyStoreConfig(
+            String keyStoreName, ServerConfigurationService serverConfiguration) throws CarbonException {
+
+        if (!KeyStoreUtil.isCustomKeyStore(keyStoreName)) {
+            throw new CarbonException("Invalid key store name. Custom key store name with the prefix expected.");
+        }
+
+        String keyStoreFileName = keyStoreName
+                .substring(RegistryResources.SecurityManagement.CustomKeyStore.KEYSTORE_PREFIX.length());
```

Comment:
Change this method to always expect custom keystore file name only
(i.e. remove the requirement of the prefix)

Reply:
Fixed in https://github.com/wso2/carbon-kernel/pull/4025/commits/938baad11ff3d0027a8299b50a1255950e925604